apps/zotonic_core/src/models/m_acl.erl

```diff
@@ -25,32 +23,63 @@
 
 %% interface functions
 -export([
-    m_find_value/3,
-    m_to_list/2,
-    m_value/2
+    m_get/2
 ]).
 
 -include_lib("zotonic.hrl").
 
+-define(is_action(A), A =:= use orelse A =:= admin orelse A =:= view
+    orelse A =:= delete orelse A =:= update orelse A =:= insert
+    orelse A =:= link).
+
+-spec m_get( list(), z:context()) -> {term(), list()}.
+m_get([ user | Rest ], Context) -> {z_acl:user(Context), Rest};
+m_get([ is_admin | Rest ], Context) -> {z_acl:is_admin(Context), Rest};
+
+% Check if current user is allowed to perform an action on some object
+m_get([ Action, Object | Rest ], Context) when ?is_action(Action), is_binary(Object) ->
+    {is_allowed_to_atom(Action, Object, Context), Rest};
+m_get([ Action, Object | Rest ], Context) when ?is_action(Action) ->
+    {z_acl:is_allowed(Action, Object, Context), Rest};
+m_get([ is_allowed, Action, Object | Rest ], Context) when ?is_action(Action), is_binary(Object) ->
+    {is_allowed_to_atom(Action, Object, Context), Rest};
+m_get([ is_allowed, Action, Object | Rest ], Context) when ?is_action(Action) ->
+    {z_acl:is_allowed(Action, Object, Context), Rest};
+
+% Check if an authenticated (default acl setttings) is allowed to perform an action on some object
+m_get([ authenticated, Action, Object | Rest ], Context) when ?is_action(Action), is_binary(Object) ->
+    {is_allowed_authenticated_to_atom(Action, Object, Context), Rest};
+m_get([ authenticated, Action, Object | Rest ], Context) when ?is_action(Action) ->
+    Context1 = case z_notifier:first(#acl_context_authenticated{}, Context) of
+                    undefined -> Context;
+                    Ctx -> Ctx
+               end,
+    {z_acl:is_allowed(Action, Object, Context1), Rest};
+m_get([ authenticated, is_allowed, Action, Object | Rest ], Context)  when ?is_action(Action), is_binary(Object) ->
+    {is_allowed_authenticated_to_atom(Action, Object, Context), Rest};
+m_get([ authenticated, is_allowed, Action, Object | Rest ], Context) when ?is_action(Action) ->
+    Context1 = case z_notifier:first(#acl_context_authenticated{}, Context) of
+                    undefined -> Context;
+                    Ctx -> Ctx
+               end,
+    {z_acl:is_allowed(Action, Object, Context1), Rest};
+
+% Error, unknown lookup.
+m_get(Vs, _Context) ->
+    lager:info("Unknown ~p lookup: ~p", [?MODULE, Vs]),
+    {undefined, []}.
```

Comment:
Why not just crash here? Is the error message too unclear?

Reply:
It is called from templates, best to give some feedback and continue.

I also used it to fix any problems in the transition (as you shouldn't get there).

Reply:
But indeed, it could be removed.

Reply:
I’m okay with keeping it there, but then we should change it to `lager:error` for proper logging.

Reply:
Done, now an error